File: hbase-to-elastic-flink/src/main/scala/com/alex/space/flink/elasticsearch/ElasticsearchSinkFunction.scala
package com.alex.space.flink.elasticsearch

import java.io.Serializable

import org.apache.flink.api.common.functions.{Function, RuntimeContext}

/**
  * @author Alex Created by Alex on 2018/9/14.
  */
trait ElasticsearchSinkFunction[T] extends Serializable with Function {
  def process(element: T, ctx: RuntimeContext, requestIndexer: RequestIndexer): Unit
}

File: flink-elasticsearch-connector/src/main/java/com/alex/space/flink/connector/ElasticsearchSinkFunction.java
package com.alex.space.flink.connector;

import java.io.Serializable;
import org.apache.flink.api.common.functions.Function;
import org.apache.flink.api.common.functions.RuntimeContext;

/**
 * @author Alex Created by Alex on 2018/9/14.
 */
public interface ElasticsearchSinkFunction<T> extends Serializable, Function {

  void process(T element, RuntimeContext ctx, RequestIndexer requestIndexer);

}
